### PR TITLE
VMS setup data migration code changes

### PR DESCRIPTION
**Pull request by NShani**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  -- under the License.
+*/
+
+
+CREATE PROCEDURE component_migrate()
+    BEGIN
+        DECLARE c int;
+        SET c = 1 ;
+        WHILE c<10 DO
+            update dojo_finding p1 set p1.sourcefilepath=(select SUBSTRING_INDEX(SUBSTRING_INDEX(SUBSTRING_INDEX(p2.description,'Vulnerable Module:',2),'Vulnerable Module:',-1),'Type:',1) from(select * from dojo_finding) p2 where sourcefilepath IS NULL order by id limit 1),p1.description=(select SUBSTRING_INDEX(p2.description,'Vulnerable Module:',1)from (select * from dojo_finding) p2 where sourcefilepath IS NULL order by id limit 1) where p1.id=(select p2.id from(select * from dojo_finding) p2 where sourcefilepath IS NULL order by id limit 1);
```
Comment (ayomawdb) [inline external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql:25]: Regarding `sourcefilepath`. 

Can't we simplify 
```
select SUBSTRING_INDEX(SUBSTRING_INDEX(SUBSTRING_INDEX(p2.description, 'Vulnerable Module:', 2), 'Vulnerable Module:', - 1), 'Type:', 1) 
```

to :::

```
select SUBSTRING_INDEX(SUBSTRING_INDEX(p2.description, 'Vulnerable Module:', -1), 'Type:', 1) 
```

Comment (ayomawdb) [inline external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql:25]: Why `where sourcefilepath IS NULL`  condition?

Comment (ayomawdb) [inline external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql:25]: Nice query.. However, I feel like relying on a NULL is a hack. Why can't we do this in one query. Example: 

```
update
   dojo_finding p1 
set
   p1.sourcefilepath = 
   (
      select
         SUBSTRING_INDEX(SUBSTRING_INDEX(p2.description, 'Vulnerable Module:', 1), 'Type:', 1) 
      from
         dojo_finding p2 
      where
         p2.id =  p1.id
   )
,
   p1.description = 
   (
      select
         SUBSTRING_INDEX(p2.description, 'Vulnerable Module:', 1)
      from
         dojo_finding p3
      where
         p3.id =  p1.id
   )
;
```

**Please test this before using. I'm just writing this on a comment, without doing any validation.**

File: external/VMS_Custom_Setup_Scripts/setup_scripts/vms_setup.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+##
+  # Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ayomawdb) [inline external/VMS_Custom_Setup_Scripts/setup_scripts/vms_setup.sh:2]: This indentation isn't the usual practice IMU

File: external/VMS_Custom_Setup_Scripts/db_migration/dojo/dojo_tools_factory.diff [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+diff --git a/dojo/tools/factory.py b/dojo/tools/factory.py
+index a24c14bb..1dff61db 100755
+--- a/dojo/tools/factory.py
++++ b/dojo/tools/factory.py
+@@ -1,4 +1,5 @@
++from dojo.tools.aa.parser import AA
```
[the remark this answers is not in the document]
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/db_migration/dojo/dojo_tools_factory.diff]: Resolved this concern in the updated PR.

File: external/VMS_Custom_Setup_Scripts/db_migration/dojo/dojo_forms.diff [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  -->
+
+diff --git a/dojo/forms.py b/dojo/forms.py
+index d7e7ecda..934827e7 100755
+--- a/dojo/forms.py
++++ b/dojo/forms.py
+@@ -297,6 +297,7 @@ class ImportScanForm(forms.Form):
+     SCAN_TYPE_CHOICES = (("", "Please Select a Scan Type"),
+                          ("Netsparker Scan", "Netsparker Scan"),
++                         ("Temp Parser", "Temp Parser"),
```
[the remark this answers is not in the document]
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/db_migration/dojo/dojo_forms.diff]: Resolved this concern in the updated PR.

File: external/VMS_Custom_Setup_Scripts/db_migration/dojo/dojo_utils.diff [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+@@ -468,22 +468,10 @@ def rename_whitesource_finding():
+ 
+ 
+@@ -544,20 +523,12 @@ def deduplication_wso2_custom(new_finding, attributes):
+             similar_findings_product = list(similar_findings_product)
+             original_finding = get_original_finding(new_finding, attributes, similar_findings_product)
+
+             if original_finding is None:
+                 product_name = new_finding.test.engagement.product.name
++                product_name = product_name.rstrip('0123456789')
```
[the remark this answers is not in the document]
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/db_migration/dojo/dojo_utils.diff]: Resolved this concern in the updated PR.

File: external/VMS_Custom_Setup_Scripts/db_migration/dojo/dojo_utils.diff [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+ 
+@@ -584,68 +555,51 @@ def deduplication_wso2_custom(new_finding, attributes):
+                             original_finding = None
+ 
+             if original_finding is not None:
+                 notes = original_finding.notes.all()
+                 new_notes = new_finding.notes.all()
+
+
++                note1 = notes.filter(note_type__name="Use Case")
```
[the remark this answers is not in the document]
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/db_migration/dojo/dojo_utils.diff]: Resolved this concern in the updated PR.

File: external/VMS_Custom_Setup_Scripts/db_migration/dojo/tools/custom_parser/parser.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+import hashlib
+import os
+
+import pandas as pd
+from tastypie import bundle
+
+from dojo.models import Finding, Notes, Note_Type, User
+
+
+class TempParser(object):
```
[the remark this answers is not in the document]
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/db_migration/dojo/tools/custom_parser/parser.py]: Resolved this concern in the updated PR.

File: external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+            SET c = c + 1 ;
+        END WHILE ;
+    END
```
[the remark this answers is not in the document]
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql]: Resolved this concern in the updated PR.

File: external/VMS_Custom_Setup_Scripts/setup_scripts/config/PR.conf [hunk 9]
```diff
@@ -0,0 +1,3 @@
+https://github.com/DefectDojo/django-DefectDojo/pull/2224.diff
+https://github.com/NShani/django-DefectDojo/pull/2.diff
```
[the remark this answers is not in the document]
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/setup_scripts/config/PR.conf:2]: Resolved this concern in the updated PR.

File: external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql (hunk 1 above)
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql]: Yes. Suggested one works well. I improved the query as suggested in the updated PR.

Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/setup_scripts/db_component_migrate_script.sql]: Yes. I missed this. This is good suggestion. I improved this suggested query with a small change and updated the PR.

File: external/VMS_Custom_Setup_Scripts/setup_scripts/vms_setup.sh (hunk 2 above)
Comment (NShani) [reply external/VMS_Custom_Setup_Scripts/setup_scripts/vms_setup.sh]: Resolved this concern in the updated PR.

File: external/VMS_DB_Backup/scripts/VmsDbBackupCon.sh [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+HOME="/backups"
+now="$(date +'%d_%m_%Y_%H')"
+fileName="db_backup_$now".gz
+backupFolder="$HOME/database_backup"
+fullPathBackupFile="$backupFolder/$fileName"
+logfile="$backupFolder/backup_log_$(date +'%Y_%m').txt"
+vmsBackupTag="[VMS Backup Process]"
+
+mysqlPassword=$(cat $HOME/scripts/config/mysqlPassword.conf)
```
Comment (ayomawdb) [inline external/VMS_DB_Backup/scripts/VmsDbBackupCon.sh:25]: One last comment. Shall we move all configs to one file:
```
mysql_username=example
mysql_password=example
```

and then use something like following to read individual value:
```
$(cat $HOME/scripts/config.conf | grep "mysql_username" | cut -d '=' -f2)
```

**Comment by ayomawdb** [comment]
@NShani File such as `external/VMS_DB_Backup/scripts/config/databaseName.conf ` (no longer required) are still in the PR. Shall we remove those, so that we can merge this?

**Review by ayomawdb — APPROVED** [review]

**Pull request merged**